### ISSUE TITLE
Cursor movement prevented when 'splitkeep' is not "cursor" and 'scrolloff'

### ISSUE THREAD

**Issue by luukvbaal** [vim/vim#12506]
### Steps to reproduce

1. Write test.vim with the following lines:
```vim
set splitkeep=screen
let &scrolloff = &lines / 4
call setline(1, range(1, 200))
wincmd s
au CursorMoved * windo echo
```
2. Run `vim --clean -S test.vim`
3. Try to move the cursor down with "j"

The cursor does not move because 'splitkeep' "fixes" the cursor position.
Any autocmd on CursorMoved that calls `win_enter_ext()` which in turn calls `win_fix_cursor()` would trigger this issue if 'scrolloff' is larger than half the window height.

(Original Neovim issue: https://github.com/neovim/neovim/issues/23956)

### Expected behaviour

The cursor is moved down.

Technically this is simply a 'scrolloff' roundoff error which might be handled directly in `win_fix_cursor()`, not sure though. Making sure that 'splitkeep' does not try to fix cursor positions that were not drawn in the first place like proposed in https://github.com/vim/vim/pull/12488 would also prevent this though (PR would need to be adjusted slightly).

### Version of Vim

VIM - Vi IMproved 9.0 (2022 Jun 28, compiled Jun  8 2023 18:35:01) Included patches: 1-1601

### Environment

Operating System: Arch Linux
Terminal: st
TERM: st-256color
SHELL: zsh

**Comment by luukvbaal** [comment]
Changing the repro steps to:
1. Write test.vim with the following lines:
```vim
set splitkeep=screen scrolloff=0
call setline(1, range(1, 200))
wincmd s
au CursorMoved * windo echo
```
2. Run `vim --clean -S test.vim`
3. Try to move the cursor to the end of the buffer with "G"

results in a related problem where "fixing" the cursor does not actually bring the cursor to the end of the buffer even without 'scrolloff' (which the proposed fix would also prevent).

**Comment by luukvbaal** [comment]
Fixed in patch 9.0.1772.
